projects/fsastorefrontlib/src/cms-components/assets-table/resolve-asset-value.pipe.ts

```diff
@@ -0,0 +1,26 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { TranslatePipe } from '@spartacus/core';
+import { RequestType } from '../../occ';
+
+@Pipe({ name: 'resolveAssetValue' })
+export class ResolveAssetValuePipe implements PipeTransform {
+  constructor(private tpipe: TranslatePipe) {}
+
+  transform(value: any, ...args: any[]) {
+    const asset = args[0];
+
+    if (value.propName) {
+      return value.value
+        .split('.')
+        .reduce((accum, curr) => accum?.[curr], asset);
+    }
+
+    const allowedFSRequestTypes = asset.categoryData.allowedFSRequestTypes;
+
+    return allowedFSRequestTypes
+      .map(allowedRequestType => allowedRequestType.requestType.code)
+      .indexOf(RequestType.FSCLAIM) > -1
+      ? this.tpipe.transform(value.value)
```

Comment:
Not sure if this is the right way to use translation in ts files. This may not work at all when switching languages. Please check this. Maybe that's why Spartacus [recommends](translatePipe) using it like an observable.